persistence/sql/persister_consent.go

```diff
@@ -135,59 +133,107 @@ func (p *Persister) GetForcedObfuscatedLoginSession(ctx context.Context, client,
 	return &s, nil
 }
 
+// CreateConsentRequest configures fields that are introduced or changed in the
+// consent request. It doesn't touch fields that would be copied from the login
+// request.
 func (p *Persister) CreateConsentRequest(ctx context.Context, req *consent.ConsentRequest) error {
-	return errorsx.WithStack(p.Connection(ctx).Create(req))
+	c, err := p.Connection(ctx).RawQuery(`
+UPDATE hydra_oauth2_flow
+SET
+	state = ?,
+	consent_challenge_id = ?,
+	consent_skip = ?,
+	consent_verifier = ?,
+	consent_csrf = ?
+WHERE login_challenge = ?;
+`,
+		flow.FlowStateConsentInitialized,
+		sqlxx.NullString(req.ID),
+		req.Skip,
+		req.Verifier,
+		req.CSRF,
+		req.LoginChallenge.String(),
+	).ExecWithCount()
+	if err != nil {
+		return sqlcon.HandleError(err)
+	}
+	if c != 1 {
+		return errorsx.WithStack(x.ErrNotFound)
+	}
```

Comment:
Wouldn't this be misleading when c == 2?

Reply:
login_challenge is the primary key and we're talking about a situation where it isn't unique. I was deciding between
1. not handling this case at all
2. handling it with a misleading error
3. handling it properly
I picked 2) because 1) felt wrong and 3) redundant. wdyt?